code/__DEFINES/callbacks.dm

```diff
@@ -1,4 +1,5 @@
 #define GLOBAL_PROC "some_magic_bullshit"
 /// A shorthand for the callback datum, [documented here](datum/callback.html)
 #define CALLBACK new /datum/callback
-#define INVOKE_ASYNC world.ImmediateInvokeAsync
+#define INVOKE_ASYNC(TRG, PR, ARG...) if ((TRG) == GLOBAL_PROC) { spawn (-1) call(PR)(##ARG); } else { spawn(-1) call(TRG, PR)(##ARG); }
+#define ASYNC spawn(-1)
```

Comment:
tf is this lol

Reply:
something i may or may not use so i dont have to explain how spawn() works in detail to contribs